.github/workflows/cron.yml

```diff
@@ -33,30 +33,55 @@ jobs:
         uses: actions/setup-python@v2
         with:
           python-version: ${{ matrix.python-version }}
-      - name: Pip cache
+
+      - name: Calculate cache metadata
+        id: cache-meta
+        shell: python
```

Comment:
I wonder can we put this script into a python file and run it instead?

Reply:
I don't think it's a good idea. Moving the context out of the workflow makes it harder to follow and forces people to look up external sources. This would be different if it was needed outside of GHA, of course.

Reply:
It would be possible to make it a composite action but it looks like an overkill.